web/src/main/java/org/mskcc/cbio/oncokb/api/pvt/PrivateSearchApiController.java

```diff
@@ -76,4 +74,240 @@
         }
         return new ResponseEntity<>(treatments, status);
     }
+
+    @Override
+    public ResponseEntity<LinkedHashSet<TypeaheadSearchResp>> searchTypeAheadGet(
+        @ApiParam(value = "The search query, it could be hugoSymbol, entrezGeneId or variant.", required = true) @RequestParam(value = "query") String query,
+        @ApiParam(value = "The limit of returned result.") @RequestParam(value = "limit", defaultValue = "5", required = false) Integer limit) {
+        LinkedHashSet<TypeaheadSearchResp> result = new LinkedHashSet<>();
+        if (limit == null) {
+            limit = TYPEAHEAD_RETURN_LIMIT;
+        }
+        if (query != null) {
+            List<String> keywords = Arrays.asList(query.trim().split("\\s+"));
+
+            if (keywords.size() == 1) {
+                // Blur search gene
+                result.addAll(convertGene(GeneUtils.searchGene(keywords.get(0), false), keywords.get(0)));
+
+                // Blur search variant
+                result.addAll(convertVariant(AlterationUtils.lookupVariant(keywords.get(0), false, AlterationUtils.getAllAlterations()), keywords.get(0)));
+            } else if (keywords.size() == 2) {
```

Comment:
should we remove this size == 2 check? what if there is space in variant name? like oncogenic mutation, A-B fusion